crates/agent/src/api/public/status.rs

```diff
@@ -9,37 +10,81 @@ use axum::{Extension, Json};
 use axum_extra::extract::Query;
 use chrono::{DateTime, Utc};
 use itertools::Itertools;
-use models::status::{self, connector::ConnectorStatus, StatusResponse};
-use models::Id;
+use models::status::{self, connector::ConnectorStatus, StatusResponse, Summary};
+use models::{CatalogType, Id};
 
 /// Query parameters for the status endpoint
 #[derive(Debug, serde::Deserialize, schemars::JsonSchema)]
 pub struct StatusQuery {
     /// The catalog name of the live spec to get the status of
     pub name: Vec<String>,
+    /// Whether to return a smaller response that excludes the `connector_status` and `controller_status` fields.
```

Comment:
meta: these are great examples of api plumbing that we wouldn't have to do w/ async-graphql